Improve intro workout in react (#1500)
enkidevs/curriculum: javascript/react/intro-react/what-is-jsx.md

---

# What is JSX?

---
## Content

> **JSX** is syntactic sugar that brings an `XML`-like syntax to **JavaScript**. Here is how it looks like:

Reviewer: I would say "HTML-like" syntax since HTML is more familiar to today's web devs than XML (and most people learning React will be web devs).

```jsx
const el = <h2>Enki!</h2>;
```

This means that **JSX** is a way of writing your **HTML** inside JS code.

A **JSX** expression is evaluated to a **React** `element`, which is short-hand for calling `React.createElement()` Think of `ReactElement`s as the **UI** building blocks of **React**. We will come back to this later.

**JSX** expressions are transformed (compiled) into standard JS objects. For example, the following **JSX** code:

```jsx
const el = <h2>Enki!</h2>;
```

will be *compiled* to:

```jsx
const el = React.createElement(
  "h2",
  null,
  "Enki!"
);
```

After performing some checks that ensure your code is bug-free, the `React.createElement()` method creates an object like this:

```jsx
const el = {
  type: 'h2',
  props: {
    children: 'Enki!'
  }
};
```

> From here, you can embed JS expressions within **JSX** by wrapping them with `{ }` curly brackets:

Reviewer: Not sure if the "from here" relates to what's directly above it.

Maybe remove it?

Reviewer: Say something like:

```suggestion
**JSX**  allows you to embed JS expressions within it wrapping them with `{ }` curly brackets:
```

```jsx
const x = "Hey there!";

const el = <h2>{x}</h2>;
```

When rendered, this element will display `"Hey there"` as an `h2`.

> **JSX** expressions can be used everywhere in JS. You can:

Reviewer: ```suggestion
Because we are using Babel to transform our entire React application, you can us **JSX** expressions anywhere. For example, you can:
```
